sympy/core/basic.py

```diff
@@ -999,32 +1000,40 @@ def fallback(self, old, new):
             Try to replace old with new in any of self's arguments.
             """
             hit = False
-            args = list(self.args)
-            for i, arg in enumerate(args):
-                if not hasattr(arg, '_eval_subs'):
-                    continue
-                arg = arg._subs(old, new, **hints)
-                if not _aresame(arg, args[i]):
-                    hit = True
-                    args[i] = arg
-            if hit:
-                rv = self.func(*args)
-                hack2 = hints.get('hack2', False)
-                if hack2 and self.is_Mul and not rv.is_Mul:  # 2-arg hack
-                    coeff = S.One
-                    nonnumber = []
-                    for i in args:
-                        if i.is_Number:
-                            coeff *= i
+            def subs_old_new(self,old,new,hit):
+                for i, arg in enumerate(args):
+                    if not hasattr(arg, '_eval_subs'):
+                        continue
+                    arg = arg._subs(old, new, **hints)
+                    if not _aresame(arg, args[i]):
+                        hit = True
+                        args[i] = arg
+                if hit:
+                    rv = self.func(*args)
+                    hack2 = hints.get('hack2', False)
+                    if hack2 and self.is_Mul and not rv.is_Mul:  # 2-arg hack
+                        coeff = S.One
+                        nonnumber = []
+                        for i in args:
+                            if i.is_Number:
+                                coeff *= i
+                            else:
+                                nonnumber.append(i)
+                        nonnumber = self.func(*nonnumber)
+                        if coeff is S.One:
+                            return nonnumber
                         else:
-                            nonnumber.append(i)
-                    nonnumber = self.func(*nonnumber)
-                    if coeff is S.One:
-                        return nonnumber
-                    else:
-                        return self.func(coeff, nonnumber, evaluate=False)
-                return rv
-            return self
+                            return self.func(coeff, nonnumber, evaluate=False)
+                    return rv
+                return self
+
+            args = list(self.args)
+            try:
+                return subs_old_new(self,old,new,hit)
+            except AttributeError:
+                if isinstance(self, Quantity):
```

Comment:
This isn't the right way to approach the problem. If Quantity class needs a subs method that works differently, then it should override _eval_subs method itself in [its code](http://docs.sympy.org/latest/_modules/sympy/physics/units/quantities.html#Quantity), not to get special treatment here